MyCryptoHQ/MyCrypto
# Parity Signer

Closes #1056

### Description

Introduces Parity Signer as another secure wallet option on the main screen. Account is unlocked by scanning a QR code of the address from the mobile app. Signing transactions is done in a modal overlaying the UI that first displays the QR code of the RLP of the transaction and then scans for the signature QR code from the mobile app.

### Changes

* New secure wallet has been added.
* New modal has been added to allow signing of the transaction while the `Promise` is being resolved.
* New reducers and actions have been added in order to pipe through the transaction from the UI to the `Promise` required by the `IFullWallet` interface.
* Added `@parity/qr-signer` as a new dependency.

### Steps to Test

1. Download the Parity Signer mobile app ([iOS](https://itunes.apple.com/us/app/parity-signer/id1218174838) / [Android](https://play.google.com/store/apps/details?id=com.nativesigner)). Create an account on it, send some ETH to the account (works with testnets).
1. Click on the "Parity Signer" option on the main screen.
1. Scan the account QR code from Parity Signer to unlock the wallet.
1. Create any kind of a transaction.
1. Sign the transaction using the app.
1. Make sure that the transaction goes through and is mined on chain.

Due to interactions with the webcam, integration tests might be tricky unless a lot of the stuff gets mocked away.

### TODO / Known issues

* [x] Needs a proper icon on the home screen.
* [x] `ParitySigner.tsx` needs to be moved to the containers.
* [x] UI needs some love (things are off-center, some paddings would be nice, some copy explaining what it is with links to the app would be nice).
* [x] Parity Signer needs a help website (linking to README.md for now...).

### Screenshots

![](http://maciej.codes/mycrypto/newicon.png)
![](http://maciej.codes/mycrypto/withlink.png)
![](http://maciej.codes/mycrypto/final3.png)
